params/genesis.go

```diff
@@ -86,14 +87,19 @@ func DefaultYoloV1GenesisBlock() *genesisT.Genesis {
 
 // DeveloperGenesisBlock returns the 'geth --dev' genesis block. Note, this must
 // be seeded with the
-func DeveloperGenesisBlock(period uint64, faucet common.Address) *genesisT.Genesis {
+func DeveloperGenesisBlock(period uint64, faucet common.Address, useEthash bool) *genesisT.Genesis {
 	// Override the default period to the user requested one
-	config := *AllCliqueProtocolChanges
-	config.Clique.Period = period
+	var config *goethereum.ChainConfig
+	if useEthash {
+		config = AllEthashProtocolChanges
```

Comment:
For the `--dev` mode to work automatically from the command line, the user has to pass `--miner.threads > 0`, I was thinking that in the case of developer chain we might want to fix the auto-mining logic by doing this check:

```
if miner.threads == 0
then setThreads = 1
```

Reply:
This seems reasonable. The whole idea of the `--dev` mode is provide a practical configuration bundle.

Reply:
Resolved by 28f9f70